src/core/api/clock.ts

```diff
@@ -345,18 +352,30 @@ export interface IClockOptions {
 function createClock(
   mediaElement : HTMLMediaElement,
   options : IClockOptions
-) : Observable<IClockTick> {
-  return observableDefer(() : Observable<IClockTick> => {
+) : { clock$: Observable<IClockTick>, setCurrentTime: (time: number) => void } {
+  let internalSeek: number | null = null;
+  const setCurrentTime = (time: number) => {
+    internalSeek = time;
+    mediaElement.currentTime = time
+  }
+  const clock$ = observableDefer(() : Observable<IClockTick> => {
     let lastTimings : IClockTick = objectAssign(
       getMediaInfos(mediaElement, "init"),
-      { stalled: null, getCurrentTime: () => mediaElement.currentTime });
+      { stalled: null,
+        getCurrentTime: () => mediaElement.currentTime,
+      });
 
     function getCurrentClockTick(state : IMediaInfosState) : IClockTick {
       const mediaTimings = getMediaInfos(mediaElement, state);
-      const stalledState = getStalledStatus(lastTimings, mediaTimings, options);
+      const stalledState = getStalledStatus(lastTimings, mediaTimings, options, internalSeek);
+      if (stalledState === null && internalSeek) {
+        internalSeek = null;
+      }
+      log.debug("API: current media element state HELLO", internalSeek);
       const timings = objectAssign({},
                                    { stalled: stalledState,
-                                     getCurrentTime: () => mediaElement.currentTime },
+                                     getCurrentTime: () => mediaElement.currentTime,
+                                    },
```

Comment:
No need for that modification here

Reply:
Done, I detached the `getCurrentTime` function of the `clock$`, so I removed that.